docs/content/guides/developer/sui-101/simulating-refs.mdx

```diff
@@ -1,6 +1,126 @@
+
+As an example, consider the following module stub that exposes an object (`Asset`) and a function (`use_asset`) to use that object.
+
+```rust
+module a_module {
+    struct Asset has key, store {
+        … // some data
+    }
+
+    public fun use_asset(asset: &Asset) {
+        …. // some code
+    }
+}
+```
+
+The function `use_asset` takes an immutable reference to the asset (`&Asset`), which is a common pattern in a transaction command. 
+
+Now consider another module that uses this asset.
+
+```rust
+module another_module {
+    struct AssetManager has key {
+        asset: Asset,
+    }
+
+    public fun get_asset(manager: &AssetManager): &Asset {
+        &manager.asset
+    }
+}
+```
+
+This module creates an object (`AssetManager`) that references the object (`Asset`) created in the previous module (`a_module`).
+
+You could then write a Move function that creates an object by reference and passes it to the `use_asset` function.  
+
+```rust
+fun do_something(manager: &AssetManager) {
+    let asset = another_module::get_asset(manager);
+    a_module::use_asset(asset);
+}
+```
+
+This function is not valid within a PTB, however, because PTBs do not support a reference returned by a function and passed to another function.
+
+To make this operation valid within a PTB, you would need to include functionality from the borrow module. Consequently, you could change the `another_module` code to the following:
+
+```rust
+module another_module {
+    struct AssetManager has key {
+        asset: Referent<Asset>,
+    }
+
+    public fun get_asset(manager: &mut AssetManager): (Asset, Borrow) {
+        borrow::borrow(&mut manager.asset)
+    }
+
+
+    public fun return_asset(
+        manager: &mut AssetManager,
+        asset: Asset,
+        b: Borrow) {
+            borrow::put_back(&mut manager.asset, asset, b)
+    }
+}
+```
+
+Now the PTB can retrieve the asset, use it in a call to `use_asset`, and return the asset.
+
+## Considerations
+
+The `Borrow` object is the key to the guarantees the borrow module offers. The definition of `Borrow` is
+`struct Borrow { ref: address, obj: ID }`
+which makes it such that you cannot drop or save its instance anywhere, so it must be consumed in the same transaction that retrieves it (hot potato). Moreover, fields in the `Borrow` struct make sure that the object returned is for the same `Referent` and the object that was originally held by the `Borrow` instance. In other words, there is no way to either keep the object retrieved or to swap it with another object in a different `Referent`.
+
+Using a reference inside an object is a very explicit and intrusive change. That has to be taken into consideration when designing a solution. Support for references in a PTB is planned, which is a much more natural and proper pattern for APIs.
```

Comment:
> Using a reference inside an object

Using a `Referent`?

the point here is that using the borrow module changes your API and data and as such is a very intrusive change.
So maybe "Support for references in PTBs ..." can go to the next line?